libbeat/otelbeat/beatconverter/beatconverter_test.go

```diff
@@ -117,34 +135,144 @@ service:
         - "filebeatreceiver"
 `
 
-func TestConverter(t *testing.T) {
-	c := converter{}
-	t.Run("test converter functionality", func(t *testing.T) {
+		input := newFromYamlString(t, unsupportedOutputConfig)
+		err := c.Convert(context.Background(), input)
+		require.ErrorContains(t, err, "output type \"kafka\" is unsupported in OTel mode")
+
+	})
+
+	t.Run("test cloud id conversion", func(t *testing.T) {
+		var supportedInput = `
+receivers:
+  filebeatreceiver:
+    filebeat:
+      inputs:
+        - type: filestream
+          enabled: true
+          id: filestream-input-id
+          paths:
+            - /tmp/flog.log
+    output:
+      elasticsearch:
+        hosts: ["https://localhost:9200"]
+        username: elastic
+        password: changeme
+        index: form-otel-exporter
+    cloud:
+      id: ZWxhc3RpYy5jbyRlcy1ob3N0bmFtZSRraWJhbmEtaG9zdG5hbWU=
+      auth: elastic-cloud:password
+service:
+  pipelines:
+    logs:
+      receivers:
+        - "filebeatreceiver"
+`
+		var expectedOutput = `
+exporters:
+  elasticsearch:
+    api_key: ""
+    endpoints:
+      - https://es-hostname.elastic.co:443
+    idle_conn_timeout: 3s
+    logs_index: form-otel-exporter
+    num_workers: 0
+    password: password
+    retry:
+      enabled: true
+      initial_interval: 1s
+      max_interval: 1m0s
+      max_retries: 3
+    user: elastic-cloud
+    timeout: 1m30s
+    batcher:
+      enabled: true
+      max_size_items: 1600
+receivers:
+  filebeatreceiver:
+    filebeat:
+      inputs:
+        - enabled: true
+          id: filestream-input-id
+          paths:
+            - /tmp/flog.log
+          type: filestream  
+    output:
+      otelconsumer: null
+    cloud: null
+    setup:
+      kibana:
+        host: https://kibana-hostname.elastic.co:443
+service:
+  pipelines:
+    logs:
+      exporters:
+        - elasticsearch
+      receivers:
+        - filebeatreceiver
+`
 
 		input := newFromYamlString(t, supportedInput)
 		err := c.Convert(context.Background(), input)
 		require.NoError(t, err, "error converting beats output config")
 
 		expOutput := newFromYamlString(t, expectedOutput)
+		compareAndAssert(t, expOutput, input)
 
-		// convert it to a common type
-		want, err := yaml.Marshal(expOutput.ToStringMap())
-		require.NoError(t, err)
-		got, err := yaml.Marshal(input.ToStringMap())
-		require.NoError(t, err)
+	})
+
+	t.Run("test local queue setting is promoted to global level", func(t *testing.T) {
```

Comment:
I'm struggling to understand this test case.
Why do we have `queue` as a part of the output configuration and then move to the `filebeatreceiver` level? It should always be on the `filebeatreceiver` level, right?

Reply:
I have the same question as @rdner, the [`queue` configuration in Beats](https://www.elastic.co/guide/en/beats/filebeat/current/configuring-internal-queue.html) is not part of the output.

Reply:
I see it now, thanks for clarifying!